src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/GRPCConnectionManager.java

```diff
@@ -177,7 +208,15 @@ private void removeAllStubs() {
   private void terminateAllConnections() {
     for (Map.Entry<String, AtomicReference<ManagedChannel>> entry : perHostChannelMap.entrySet()) {
       LOG.debug("shutting down connection to host: {}", entry.getKey());
-      entry.getValue().get().shutdownNow();
+      ManagedChannel channel = entry.getValue().get();
+      channel.shutdownNow();
+      try {
+        channel.awaitTermination(1, TimeUnit.MINUTES);
+      } catch (InterruptedException e) {
```

Comment:
You need to propagate the interrupted status when catching an InterruptedException by calling Thread.currentThread().interrupt() after channel.shutdownNow();

Reply:
Done